src/textual/app.py

```diff
@@ -1081,6 +1082,14 @@ async def action_bang(self) -> None:
     async def action_bell(self) -> None:
         self.bell()
 
+    async def action_focus(self, widget_id: str) -> None:
+        try:
+            node = self.query(f"#{widget_id}").first()
+            if isinstance(node, Widget):
+                self.set_focus(node)
+        except NoMatchingNodesError:
+            pass
```

Comment:
We could maybe shrink the scope of the `try / except` block, and replace it with an _early return_?
(but that's very subjective of course, so feel free to ignore this comment :smile: )
```suggestion
        try:
            node = self.query(f"#{widget_id}").first()
        except NoMatchingNodesError:
            return
        if isinstance(node, Widget):
            self.set_focus(node)
```

Reply:
I'd use the else: clause here. I too like the smaller scope for the except block

```suggestion
        try:
            node = self.query(f"#{widget_id}").first()
        except NoMatchingNodesError:
            pass
        else:
            if isinstance(node, Widget):
                self.set_focus(node)
```